### PR TITLE
[WIP] Dropout sched exp

### PR DESCRIPTION
**Pull request by DylanASHillier**
See [prereport](pre_reports/dropout_prereport.pdf) for more detail

### CODE REVIEW

**Comment by DylanASHillier** [comment]
request that we merge in this current version as it fixes things we are all facing. but still its freezing...

File: trainers/datasets.py [hunk 1]
```diff
@@ -56,15 +56,6 @@ def __len__(self):
     
     def __getitem__(self, idx):
         raise NotImplementedError
-    
-    def check_processed(self):
```
Comment (DylanASHillier) [inline trainers/datasets.py:60]: doesn't do anything...

File: trainers/utils.py [hunk 2]
```diff
@@ -310,9 +310,9 @@ def print_evaluation_results(iter_num, eval_results, benchmark_results):
         for benchmark, value in benchmark_results[eval_method].items():
             benchmark_table.add_row([
                 f"{benchmark}", 
-                value['accuracy'].item(),
```
Comment (DylanASHillier) [inline trainers/utils.py:313]: bug

File: train.py [hunk 3]
```diff
@@ -33,15 +34,12 @@ def ddp_main(rank, world_size, cfg):
...
         # load the relevant trainer
-        trainer = build_trainer(
+        trainer: base_trainer.BaseTrainer = build_trainer(
             cfg=cfg,
             model=model,
             gpu_id=rank
         )
         print(f"Rank{rank} Trainer built")
-        # preprocess the training data
-        trainer.preprocess_data()
```
Comment (DylanASHillier) [inline train.py:43]: no longer extant

File: trainers/datasets.py (hunk 1 above)
Comment (LeonGuertler) [reply trainers/datasets.py:60]: I think it is worth keeping. When people init the dataset without it being preprocessed we should throw an error.

**Review by LeonGuertler — COMMENTED** [review]
mostly looks good

File: trainers/datasets.py (hunk 1 above)
Comment (DylanASHillier) [reply trainers/datasets.py:60]: it literally isn't called anywhere. fine to add it into the init but rn it doesn't do anything

File: trainers/datasets.py [hunk 4]
```diff
@@ -42,6 +42,8 @@ def _load_data(self):
         """
         Get data
         """
+        if not os.path.exists(self.data_path):
```
Comment (DylanASHillier) [inline trainers/datasets.py:45]: think this resolves

Comment (LeonGuertler) [reply trainers/datasets.py:45]: good solution

**Pull request merged**